### DISCUSSION TITLE
[Key]

### DISCUSSION THREAD

**Discussion by etaptecnologia** [balta-io/forum#163] [Tira Dúvidas] [answered]
Boa tarde comunidade Balta.

No código abaixo existe um [key], eu assisti todas as aulas desse módulo e não encontrei a aula que fala sobre isso.

`namespace Blog.Models
{
    [Table("[User]")]
    public class User
    {
        public User()
        {
            Roles = new List<Role>();
        }

        [Key]  <------------------- esse aqui
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string PasswordHash { get; set; }
        public string Bio { get; set; }
        public string Image { get; set; }
        public string Slug { get; set; }

        [Write(false)] 
        public List<Role> Roles { get; set; }
    `

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @etaptecnologia, conforme mostrado no curso de Dapper, o atributo `Key` define que a propriedade é a chave primária da tabela.

**Reply by etaptecnologia** [reply]
@andrebaltieri Boa tarde, voce poderia por favor me mostrar em qual Aula do modulo sobre Dapper voce diz isso? eu assisti todas as aulas e nao encontrei.

**Reply by DiegoOliver94** [reply]
@etaptecnologia no curso 
"Fundamentos do Entity Framework" no modulo "Operações básicas" na aula " Table, Key e Identity" ele explica o porquê do uso da "Key"

**Reply by andrebaltieri** [reply]
Obrigado @DiegoOliver94 💜

**Reply by etaptecnologia** [reply]
Muito obrigado, eu nao encontrei pois nao fiz as aulas de EF fui diretamente para a aula de Dapper.
